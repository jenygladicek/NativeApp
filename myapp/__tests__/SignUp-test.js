import React, {Component} from 'react';
import  'react-native';
import SignUpCompo from './../src/components/SignUp';

import renderer from 'react-test-renderer';
import {configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import { shallow, mount, render } from 'enzyme';

describe('Signup Component', () => {
    it('Check the render of signup component', () => {
        expect(shallow(<SignUpCompo />).exists()).toBe(true)
    });
    it('checking whether renders without change by matching previous snapshot', () => {
        const component = renderer.create(<SignUpCompo />);
        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });
    it('renders a mobile number input', () => {
        expect(shallow(<SignUpCompo />).find('#mobNo').length).toEqual(1)
    });
    describe('Mobile Number input', () => {
  
        it('should respond to change event and change the state of the Singup Component', () => {
         
         const wrapper = shallow(<SignUpCompo />);
         wrapper.find('#mobNo').simulate('change', {target: {name: 'text', value: ''}});
         
        expect(wrapper.state('text')).toEqual('');
        })
       })
    
  });