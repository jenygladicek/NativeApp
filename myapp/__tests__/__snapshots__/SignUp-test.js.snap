// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Component checking whether renders without change by matching previous snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <Text
    style={
      Object {
        "color": "grey",
        "fontSize": 15,
      }
    }
  >
    Mobile Number
  </Text>
  <TextInput
    allowFontScaling={true}
    id="mobNo"
    keyboardType="numeric"
    name="text"
    onChangeText={[Function]}
    style={
      Object {
        "borderBottomColor": "grey",
        "borderBottomWidth": 1,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 20,
        "marginTop": 20,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignSelf": "flex-start",
            "borderColor": "red",
            "borderWidth": 1,
          },
          undefined,
        ]
      }
    />
    <Text
      style={
        Object {
          "color": "grey",
          "fontSize": 15,
        }
      }
    >
      I agree to Stellapps Terms of Use and Privacy Policy
    </Text>
  </View>
  <View
    accessibilityRole="button"
    accessibilityStates={
      Array [
        "disabled",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {},
        ]
      }
    >
      <Text
        disabled={true}
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#009887",
            },
            Object {
              "color": "#cdcdcd",
            },
          ]
        }
      >
        Sign Up
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": 20,
        "marginTop": 20,
      }
    }
  >
    <Text
      onPress={[Function]}
      style={
        Object {
          "color": "grey",
          "fontSize": 15,
        }
      }
    >
      Already Member? Sign In
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": 20,
        "marginTop": 20,
      }
    }
  >
    <Text />
  </View>
</View>
`;
